src/lib.rs

```diff
@@ -111,6 +111,29 @@ pub struct Acquired {
     data: imp::Acquired,
 }
 
+impl Acquired {
+    /// This drops the `Acquired` token without releasing the associated token.
+    ///
+    /// This is not generally useful, but can be helpful if you do not have the
+    /// ability to store an Acquired token but need to not yet release it.
+    pub fn drop_without_releasing(self) {
+        // We currently cannot move out of structs with Drop impls so just move
+        // the fields out and then forget the struct.
+        let (client, data) = unsafe {
+            let Acquired {
+                ref client,
+                ref data,
+            } = &self;
+            let client = std::ptr::read(client);
+            let data = std::ptr::read(data);
+            std::mem::forget(self);
+            (client, data)
```

Comment:
I don't know what a good way to implement this is to be honest :)

I'm happy to also just add a bool to the struct and gate on that in the drop impl if that seems better.